libp2p/errors.nim

```diff
@@ -37,3 +37,14 @@ macro checkFutures*[T](futs: seq[Future[T]], exclude: untyped = []): untyped =
             # We still don't abort but warn
             warn "Something went wrong in a future",
                 error=exc.name, file=pos.filename, line=pos.line
+
+proc allFuturesThrowing*[T](args: varargs[Future[T]]): Future[void] =
```

Comment:
we should probably switch to `allFuturesThrowing ` everywhere?

Reply:
Already done except, `directchat` and `testinterop` IIRC (maybe some more minor location), should we swap those too?
I guess they all assumed exception throwing right?